src/Controller/OgAdminMembersController.php

```diff
@@ -21,13 +53,117 @@ class OgAdminMembersController extends ControllerBase {
    *   The members overview View.
    */
   public function membersList(RouteMatchInterface $route_match) {
-    $parameter_name = $route_match->getRouteObject()->getOption('_og_entity_type_id');
-
     /** @var \Drupal\Core\Entity\EntityInterface $group */
-    $group = $route_match->getParameter($parameter_name);
+    $group = $this->getGroupFromRoute($route_match);
 
     $arguments = [$group->getEntityTypeId(), $group->id()];
     return Views::getView('og_members_overview')->executeDisplay('default', $arguments);
   }
 
+  /**
+   * Displays add membership links for available membership types.
+   *
+   * Returns default membership type if that's all that exists.
+   *
+   * @return array
+   *   A render array for a list of the node types that can be added; however,
+   *   if there is only one node type defined for the site, the function
+   *   will return the default add member form.
+   */
+  public function addPage(RouteMatchInterface $route_match) {
+    $membership_type_list = $this->entityTypeManager()->getStorage('og_membership_type')->loadMultiple();
+    if ($membership_type_list && count($membership_type_list) == 1) {
+      return $this->addForm($route_match);
+    }
+
+    $build = [
+      '#theme' => 'entity_add_list',
+      '#bundles' => [],
+    ];
+
+    $bundle_entity_type = $this->entityTypeManager()->getDefinition('og_membership_type');
+    $build['#cache']['tags'] = $bundle_entity_type->getListCacheTags();
+
+    $group = $this->getGroupFromRoute($route_match);
+
+    $add_link_params = [
+      'group' => $group->id(),
+      'entity_type_id' => $group->getEntityType()->id(),
+    ];
+
+    foreach ($membership_type_list as $membership_type_id => $og_membership_type) {
+      $add_link_params['membership_type'] = $membership_type_id;
+      $build['#bundles'][$membership_type_id] = [
+        'label' => $og_membership_type->label(),
+        'add_link' => Link::createFromRoute($og_membership_type->label(), "entity.og_membership.add_form", $add_link_params),
+      ];
+    }
+
+    return $build;
+  }
+
+  /**
+   * Provides the add member submission form.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route match service.
+   *
+   * @return array
+   *   The member add form.
+   */
+  public function addForm(RouteMatchInterface $route_match) {
+    /** @var \Drupal\og\Entity\OgMembershipType $membership_type */
+    $membership_type = $route_match->getParameter('membership_type');
+    if (is_object($membership_type)) {
+      $membership_type = $membership_type->id();
+    }
+    if ($membership_type === NULL) {
+      $membership_type = 'default';
```

Comment:
@jludwig 
\Drupal\og\OgMembershipInterface::TYPE_DEFAULT exists, would be nicer to have that.